### DISCUSSION TITLE
How to rename a variable?

### DISCUSSION THREAD

**Discussion by coffent** [rubberduck-vba/Rubberduck#6104] [Q&A] [answered]
I wanted a way to safely rename all instances of a variable, as can be done in Visual Basic.  I found Rubberduck & installed it.  If I highlight a variable that I want to rename, right-click, select Rubberduck then Refactor, all options including Rename are greyed out.  What do I need to do?  Thanks.

**Comment by retailcoder** [comment, marked as answer]
Hi, thanks for reaching out! Rubberduck needs to parse the code and make (and maintain) an internal representation of the code before it can do ...most of the things it can do 😄 

There should be a new "Rubberduck" toolbar in the IDE; the left-most label is actually a button, says "Pending" at first, but then you click it and it goes through a number of states before it stops at "Ready" - at that point everything should be enabled everywhere.

Whenever you modify a module and stop to re-read, consider refreshing (there's a keyboard shortcut you can configure for that, default is <kbd>Ctrl</kbd>+<kbd>`</kbd>); the fewer modules need to be re-processed, the faster re-resolving references completes.

**Reply by coffent** [reply]
Got it - thanks!
